smithy-aws-protocol-tests/model/restJson1/malformedRequests/malformed-content-type.smithy

```diff
@@ -0,0 +1,137 @@
+$version: "1.0"
+
+namespace aws.protocoltests.restjson
+
+use aws.protocols#restJson1
+use smithy.test#httpMalformedRequestTests
+use aws.protocoltests.shared#GreetingStruct
+use aws.protocoltests.shared#JpegBlob
+
+apply MalformedContentTypeWithoutBody @httpMalformedRequestTests([
+    {
+        id: "RestJsonWithoutBodyExpectsEmptyContentType",
+        documentation: """
+        When there is no modeled input, content type must not be set and the body must be empty.""",
+        protocol: restJson1,
+        request: {
+            method: "POST",
+            uri: "/MalformedContentTypeWithoutBody",
+            body: "{}",
+            headers: {
+                // this should be omitted
+                "content-type": "application/json"
+            }
+        },
+        response: {
+            code: 415,
+            headers: {
+                "x-amzn-errortype": "UnsupportedMediaTypeException"
+            }
+        },
+        tags: [ "content-type" ]
+    }
+])
+
+apply MalformedContentTypeWithBody @httpMalformedRequestTests([
+    {
+        id: "RestJsonWithBodyExpectsApplicationJsonContentType",
+        documentation: """
+        When there is modeled input, they content type must be application/json""",
+        protocol: restJson1,
+        request: {
+            method: "POST",
+            uri: "/MalformedContentTypeWithBody",
+            body: "{}",
+            headers: {
+                // this should be application/json
+                "content-type": "application/hal+json"
+            }
+        },
+        response: {
+            code: 415,
+            headers: {
+                "x-amzn-errortype": "UnsupportedMediaTypeException"
+            }
+        },
+        tags: [ "content-type" ]
+    }
+])
+
+apply MalformedContentTypeWithPayload @httpMalformedRequestTests([
+    {
+        id: "RestJsonWithPayloadExpectsModeledContentType",
+        documentation: """
+        When there is a payload with a mediaType trait, the content type must match.""",
+        protocol: restJson1,
+        request: {
+            method: "POST",
+            uri: "/MalformedContentTypeWithPayload",
+            body: "{}",
+            headers: {
+                // this should be image/jpeg
+                "content-type": "application/json"
```

Comment:
`payload` below doesn't have an indicated mediaType, shouldn't this pass?

Reply:
The payload is `JpegBlob` with has a media type `image/jpeg`